Dockerfile

```diff
@@ -0,0 +1,88 @@
+# syntax=docker/dockerfile:1
+
+ARG CUDA_VER=11.8.0
+ARG PYTHON_VER=3.10
+ARG LINUX_VER=ubuntu22.04
+
+ARG RAPIDS_VER=23.08
+ARG DASK_SQL_VER=2023.6.0
+
+ARG BASE_FROM_IMAGE=rapidsai/mambaforge-cuda
+
+# Gather dependency information
+FROM rapidsai/ci:latest AS dependencies
+ARG CUDA_VER
+ARG PYTHON_VER
+
+ARG RAPIDS_VER
+
+ARG RAPIDS_BRANCH="branch-${RAPIDS_VER}"
+
+RUN pip install --upgrade conda-merge rapids-dependency-file-generator
+
+COPY notebooks.sh /notebooks.sh
+
+RUN /notebooks.sh
+
+
+# Base image
+FROM ${BASE_FROM_IMAGE}:cuda${CUDA_VER}-base-${LINUX_VER}-py${PYTHON_VER} as base
+ARG CUDA_VER
+ARG PYTHON_VER
+
+ARG RAPIDS_VER
+ARG DASK_SQL_VER
+
+RUN useradd -rm -d /home/rapids -s /bin/bash -g conda -u 1001 rapids
+
+USER rapids
+
+WORKDIR /home/rapids
+
+COPY condarc /opt/conda/.condarc
+
+RUN mamba install -y -n base \
+        "rapids=${RAPIDS_VER}.*" \
+        "dask-sql=${DASK_SQL_VER%.*}.*" \
+        "python=${PYTHON_VER}.*" \
+        "cuda-version=${CUDA_VER%.*}.*" \
+        ipython \
+    && conda clean -afy
+
+COPY entrypoint.sh /home/rapids/entrypoint.sh
+
+ENTRYPOINT ["/home/rapids/entrypoint.sh"]
+
+CMD ["ipython"]
+
+
+# Notebooks image
+FROM base as notebooks
```

Comment:
Do we like having multiple images in a Dockerfile? I rarely see it done, although I do see some advantages here.

Reply:
Gotcha. Yeah I'm familiar with the possibility but didn't know that was an intentional design choice here.